cacheops/utils.py

```diff
@@ -168,6 +168,24 @@ def wrapper(request, *args, **kwargs):
     return cached_view
 
 
+import datetime
+from contextlib import contextmanager
+
+def elapser(start):
+    # Returns the elapsed time in milliseconds
+    # https://github.com/antirez/redis/blob/unstable/src/server.c#L396
+    now = datetime.datetime.utcnow() - start
+    m = now.seconds * 1000000
+    m += now.microseconds
+    return m
+
+# http://stackoverflow.com/questions/7370801/measure-time-elapsed-in-python
+
+@contextmanager
+def elapsed_timer():
+    start = datetime.datetime.utcnow()
+    yield lambda: elapser(start)
+
```

Comment:
This is definitely a cool context manager.

The comment link has some code that freezes the elapsed time when exiting the context manager. And while I don't fully grok how this happens, it looks to me that the elapser lambda will return the elapsed time from entering context manager to the time where it's called (not the exit of the context manager).

Tests would be awesome if it's reasonable to code up.